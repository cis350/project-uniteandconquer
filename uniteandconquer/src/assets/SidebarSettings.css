.sideBarSettings {
  margin-bottom: 10px;
}

.greeting {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 10px;

  position: absolute;
  width: 303px;
  height: 98px;
  left: 51px;
  top: 50px;

  background: #fae9c7;
  border: 5px solid #ffd9a0;
  box-sizing: border-box;
}

.link {
  text-decoration: none;
  color: black;
}

.main-page {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 10px;

  position: absolute;
  width: 303px;
  height: 98px;
  left: 51px;
  top: 167px;

  background: #fae9c7;
  border: 16px solid #ffd9a0;
  box-sizing: border-box;

  white-space: nowrap;
}

.main-page:hover {
  background-color: #ffd9a0;
}

.text {
  padding-left: 130px;
  padding-right: 130px;
  padding-top: 30px;
  padding-bottom: 30px;
}

.notification {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 10px;

  position: absolute;
  width: 303px;
  height: 98px;
  left: 51px;
  top: 289px;

  background: #fae9c7;
  border: 16px solid #ffd9a0;
  box-sizing: border-box;
}

.notification:hover {
  background-color: #ffd9a0;
}

.personal-information {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 10px;

  position: absolute;
  width: 303px;
  height: 98px;
  left: 51px;
  top: 412px;

  background: #fae9c7;
  border: 16px solid #ffd9a0;
  box-sizing: border-box;
}

.personal-information:hover {
  background-color: #ffd9a0;
}

.password {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 10px;

  position: absolute;
  width: 303px;
  height: 100px;
  left: 51px;
  top: 535px;
  bottom: 635px;

  background: #fae9c7;
  border: 16px solid #ffd9a0;
  box-sizing: border-box;
}

.password:hover {
  background-color: #ffd9a0;
}
