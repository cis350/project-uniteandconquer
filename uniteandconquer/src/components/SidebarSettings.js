import React from 'react';
import { Link } from 'react-router-dom';
import '../assets/SidebarSettings.css';

const myStorage = window.sessionStorage;
function SidebarSettings() {
  return (
    <div className="sidebarSettings">
      <div className="greeting">
        <i className="far fa-user-circle fa-2x" />
        {' '}
        Hi,
        {myStorage.getItem('firstName') ? myStorage.getItem('firstName') : 'Guest'}
      </div>

      <div data-testid="main-page-header" className="main-page">
        <Link className="link" to="/">
          <div className="text">Back to Main Page</div>
        </Link>
      </div>
      <div data-testid="personal-information-header" className="personal-information">
        <Link className="link" to="/user-settings-personal-information">
          <div className="text">Personal Information</div>
        </Link>
      </div>
      <div data-testid="password-header" className="password">
        <Link className="link" to="/user-settings-password">
          <div className="text">Password</div>
        </Link>
      </div>
    </div>
  );
}

export default SidebarSettings;
