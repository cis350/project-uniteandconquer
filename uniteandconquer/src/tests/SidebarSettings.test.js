/**
 * @jest-environment jsdom
 */

import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import {
  render, screen,
} from '@testing-library/react';
import SidebarSettings from '../components/SidebarSettings';

// https://stackoverflow.com/questions/66284286/react-jest-mock-usenavigate

const mockedUsedLink = jest.fn();
const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  Link: () => mockedUsedLink,
  useNavigate: () => mockedUsedNavigate,
}));

test('sidebar header (main/home page) content', () => {
  render(<SidebarSettings />);
  const headerElement = screen.getByTestId('main-page-header');
  expect(headerElement).toBeInTheDocument();
});

test('sidebar header (Personal Information) content', () => {
  render(<SidebarSettings />);
  const headerElement = screen.getByTestId('personal-information-header');
  expect(headerElement).toBeInTheDocument();
});

test('sidebar header (Password) content', () => {
  render(<SidebarSettings />);
  const headerElement = screen.getByTestId('password-header');
  expect(headerElement).toBeInTheDocument();
});

test('snapshot sidebar for settings', () => {
  const component = renderer.create(<SidebarSettings />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
