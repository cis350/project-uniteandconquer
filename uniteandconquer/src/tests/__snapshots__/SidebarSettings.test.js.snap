// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot sidebar for settings 1`] = `
<div
  className="sidebarSettings"
>
  <div
    className="greeting"
  >
    <i
      className="far fa-user-circle fa-2x"
    />
     
    Hi,
    Guest
  </div>
  <div
    className="main-page"
    data-testid="main-page-header"
  />
  <div
    className="personal-information"
    data-testid="personal-information-header"
  />
  <div
    className="password"
    data-testid="password-header"
  />
</div>
`;
